### ISSUE TITLE
MapDataset.to_image() fails when keepdims=False

### ISSUE THREAD

**Issue by Bultako** [gammapy/gammapy#2537]
I have a `KeyError` exception trying to get 2D Maps from a stacked Dataset.

```
for obs in observations:
    dataset = maker.run(obs)
    stacked.stack(dataset)
dataset_2d = stacked.to_image(spectrum=spectrum, keepdims=False)

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-48-8ff62ab62aa3> in <module>
      1 spectrum = PowerLaw2SpectralModel(index=2)
----> 2 dataset_2d = stacked.to_image(keepdims=False)
      3 
      4 maps2D = {
      5     "counts": dataset_2d.counts,

~/git/gammapy/gammapy/gammapy/cube/fit.py in to_image(self, spectrum, keepdims)
    903             counts=counts,
    904             exposure=exposure,
--> 905             background_model=BackgroundModel(background),
    906             mask_safe=mask_image,
    907             edisp=edisp,

~/git/gammapy/gammapy/gammapy/modeling/models/cube.py in __init__(self, map, norm, tilt, reference, name, filename)
    419         filename=None,
    420     ):
--> 421         axis = map.geom.get_axis_by_name("energy")
    422         if axis.node_type != "edges":
    423             raise ValueError('Need an integrated map, energy axis node_type="edges"')

~/git/gammapy/gammapy/gammapy/maps/geom.py in get_axis_by_name(self, name)
   1522         """
   1523         axes = {axis.name.upper(): axis for axis in self.axes}
-> 1524         return axes[name.upper()]
   1525 
   1526     def get_axis_index_by_name(self, name):

KeyError: 'ENERGY'
```

```
stacked.background_model.map

WcsNDMap

	geom  : WcsGeom 
 	axes  : ['lon', 'lat', 'energy']
	shape : (500, 400, 10)
	ndim  : 3
	unit  : 
	dtype : float64
```

**Comment by adonath** [comment]
Thanks @Bultako, indeed this is a problem. Right now we don't have a `BackgroundModel`, that supports maps without an energy axis. My proposal here would be to actually remove the `keepdims` option, because we support image based analysis with a `MapDataset` and one energy bin.
 
Do you need `keepdims=False` for a specific use-case?

**Comment by Bultako** [comment]
Ok to remove `keepdims`.

The use case is to extend the high-level interface with source-detection features.
I wanted to use `TSMapEstimator.run()` which need 2D maps as input, and I was trying to produce 2D maps from a stacked dataset with `MapDataset.to_image(keepdims=False)`. I guess I could use `sum_over_axes()` method to remove the energy axis of the produced maps.

**Comment by adonath** [comment]
Resolved by #2637
